src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -213,7 +213,7 @@ internal List<PluginFile> GetPluginsInNuGetPluginPaths()
                         pluginFiles.AddRange(GetNetToolsPluginsInDirectory(path) ?? new List<PluginFile>());
                     }
                 }
-                else
+                else if (!string.IsNullOrEmpty(path))
```

Comment:
I would move this logic to line 190 to just skip the entry altogether if its null or empty.  Alternatively, you can pass `StringSplitOptions.RemoveEmptyEntries` to String.Split()

```c#

string[] paths = _nuGetPluginPaths?.Split(new[] { Path.PathSeparator }, StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
```